File: 0009-palindrome-number/0009-palindrome-number.py
class Solution(object):
    def isPalindrome(self, x):
        if x < 0 or (x % 10 == 0 and x != 0):
            return False
        reversed_half = 0
        while x > reversed_half:
            reversed_half = reversed_half * 10 + x % 10
            x //= 10
        return x == reversed_half or x == reversed_half // 10
        # if x<0:return False
        # x=str(x)
        # l=0
        # r=len(x)-1
        # while l<r:
        #     if x[l] != x[r]:
        #         return False
        #     l += 1
        #     r -= 1
        # return True

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution 
{
    public boolean isPalindrome(int x) 
    {
        if(x<0)
            return false;
        int y = x;
        int num=0;
        while(y>0)
        {
            num = 10 * num + (y%10);
            y /= 10;
        }
        return num == x;
    }
}





















// class Solution 
// {
//     public boolean isPalindrome(int x) 
//     {
//         if (x < 0 || (x % 10 == 0 && x != 0)) 
//         {
//             return false;
//         }
//         int reversedHalf = 0;
//         while (x > reversedHalf) 
//         {
//             reversedHalf = reversedHalf * 10 + x % 10;
//             x /= 10;
//         }
//         return x == reversedHalf || x == reversedHalf / 10;
//     }
// }
